functions/internal/runner/handle_tool_call.go

```diff
@@ -157,12 +165,37 @@ func (s *Service) executeRequest(ctx context.Context, req callRequest, w http.Re
 			w.Header().Add(key, value)
 		}
 	}
+
+	// Announce trailers for resource usage metrics
+	// We currently have to write these after WriteHeader
+	w.Header().Set("Trailer", cpuHeader+", "+memoryHeader+", "+executionTimeHeader)
+
 	w.WriteHeader(response.StatusCode)
 	if _, err := io.Copy(w, response.Body); err != nil {
 		s.logger.ErrorContext(ctx, "failed to copy response body", attr.SlogError(err))
 	}
 
 	err = cmd.Wait()
+	executionTime := time.Since(startTime).Seconds()
+
+	// Write resource usage as trailers after response body is sent
+	w.Header().Set(executionTimeHeader, fmt.Sprintf("%.2f", executionTime))
```

Comment:
This would result in loss of precision no? Is `%.17g` for float64 values the right call here?